pysatMissions/instruments/missions_sgp4.py

```diff
@@ -118,93 +169,161 @@ def load(fnames, tag=None, inst_id=None, obs_long=0., obs_lat=0., obs_alt=0.,
+                           mean_motion, np.radians(raan))
+    else:
+        # Otherwise, use TLEs
+        satellite = sapi.Satrec.twoline2rv(line1, line2, sapi.WGS72)
+        mean_motion = satellite.mm
+
+    if one_orbit:
+        ind = times <= (2 * np.pi / mean_motion * 60)
+        times = times[ind]
+        index = index[ind]
+
+    jd = jd * np.ones(len(times))
+    fr = times / 86400.
+
+    err_code, position, velocity = satellite.sgp4_array(jd, fr)
+
+    # Check all propagated values for errors in propagation
+    errors = np.unique(err_code[err_code > 0])
+    if len(errors) > 0:
+        # Raise highest priority error.
+        raise ValueError(sapi.SGP4_ERRORS[errors[0]])
+
+    # Add ECEF values to instrument.
+
+    pos_ecef = conv_sph.eci2ecef(position, (jd + fr))
+    vel_ecef = conv_sph.eci2ecef(velocity, (jd + fr))
+
+    # Convert to geocentric latitude, longitude, altitude.
+    lat, lon, rad = conv_sph.ecef_cart2spherical(pos_ecef)
+    # Convert to geodetic latitude, longitude, altitude.
+    # Ellipsoidal conversions require input in meters.
+    geod_lat, geod_lon, geod_alt = conv_ell.ecef_cart2geodetic(pos_ecef * 1000.)
 
     # Put data into DataFrame
-    data = pds.DataFrame({'position_eci_x': position[::3],
-                          'position_eci_y': position[1::3],
-                          'position_eci_z': position[2::3],
-                          'velocity_eci_x': velocity[::3],
-                          'velocity_eci_y': velocity[1::3],
-                          'velocity_eci_z': velocity[2::3]},
+    data = pds.DataFrame({'position_eci_x': position[:, 0],
+                          'position_eci_y': position[:, 1],
+                          'position_eci_z': position[:, 2],
+                          'velocity_eci_x': velocity[:, 0],
+                          'velocity_eci_y': velocity[:, 1],
+                          'velocity_eci_z': velocity[:, 2],
+                          'position_ecef_x': pos_ecef[:, 0],
+                          'position_ecef_y': pos_ecef[:, 1],
+                          'position_ecef_z': pos_ecef[:, 2],
+                          'velocity_ecef_x': vel_ecef[:, 0],
+                          'velocity_ecef_y': vel_ecef[:, 1],
+                          'velocity_ecef_z': vel_ecef[:, 2],
+                          'latitude': lat,
+                          'longitude': lon,
+                          'mean_altitude': rad - 6371.2,
+                          'geod_latitude': geod_lat,
+                          'geod_longitude': geod_lon,
+                          'geod_altitude': geod_alt / 1000.},  # Convert to km
                          index=index)
     data.index.name = 'Epoch'
 
-    # TODO: add call for GEI/ECEF translation here => #56
-
-    return data, meta.copy()
+    # Create metadata corresponding to variables in load routine
+    meta = pysat.Meta()
+    meta['Epoch'] = {
+        meta.labels.units: 'Milliseconds since 1970-1-1',
```

Comment:
Strictly this isn't true once loaded into a datetime. The note about the UTC time at middle of measurement is a good one. The rest should be removed. The lates `load_meta_proc` in pysat drops meta data for time index. The note about the middle measurement is the fly in that ointment as it is pertinent.

Could add it to all the variables, though repetitive.....